src/object/srv_obj.c

```diff
@@ -954,6 +950,24 @@ obj_local_rw(crt_rpc_t *rpc, struct ds_cont_hdl *cont_hdl,
 		goto out;
 	}
 
+	if (obj_rpc_is_fetch(rpc) && !size_fetch) {
+		obj_fetch_csum_init(cont_hdl, orw, orwo);
+
+		obj_fetch_csums_link(orw, orwo);
+
+		rc = vos_fetch_csum(ioh,
+				    orw->orw_iods.ca_arrays,
+				    orw->orw_iods.ca_count,
+				    cont_hdl->sch_csummer);
+		obj_fetch_csums_unlink(orw);
+
+		if (rc) {
+			D_ERROR(DF_UOID" fetch verify failed: %d.\n",
+				DP_UOID(orw->orw_oid), rc);
+			goto post;
+		}
+	}
+
 	if (rma) {
```

Comment:
So bulk transfer will mistakenly use the 'raw buf'? I think bio_sgl_convert() needs be changed to use requested buf.

Reply:
fixed